org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/lsp/ClosingTagReconcilingStrategy.java

```diff
@@ -0,0 +1,115 @@
+package org.eclipse.wildwebdeveloper.xml.internal.lsp;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.reconciler.DirtyRegion;
+import org.eclipse.jface.text.reconciler.IReconcilingStrategy;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.lsp4e.LanguageServiceAccessor;
+import org.eclipse.lsp4e.LanguageServiceAccessor.LSPDocumentInfo;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.wildwebdeveloper.xml.internal.XMLLanguageServerAPI;
+
+public class ClosingTagReconcilingStrategy implements IReconcilingStrategy {
+
+
+	private ITextViewer viewer;
+
+	public void documentChanged(DocumentEvent e) {
+		update(e);	
+	}
+
+	public void update(DocumentEvent event) {
+		if (event == null || viewer == null) {
+			return;
+		}
+		IDocument document = event.getDocument();
+		if (document == null || event == null || event.getLength() != 0 || event.getText().length() != 1) {
+			return;
+		}
+
+		int offset = event.getOffset() + 1;
+		char c = event.getText().charAt(0);
+		if (c != '>' && c != '/') {
+			return;
+		}
+		URI uri = LSPEclipseUtils.toUri(document);
+		if (uri == null) {
+			return;
+		}
+
+		TextDocumentIdentifier identifier = new TextDocumentIdentifier(uri.toString());
+		Optional<LSPDocumentInfo> info = LanguageServiceAccessor
+				.getLSPDocumentInfosFor(document, (capabilities) -> true).stream()
+				.filter(doc -> (doc.getLanguageClient() instanceof XMLLanguageServerAPI)).findAny();

```

Comment:
@mickaelistria is it the right mean to get the XML language server to consume after the custom `xml/closeTag` request?

Reply:
That should work. I believe there are some kind of APIs in LSP4J to allow introspecting a language server to check whether it has the corresponding JsonRequest support, but I don't think that would be simpler and maybe not even better...
So sticking to it is fine as long as it works.
By the way, a test to ensure it works would be welcome.

Reply:
I write a tests autoCloseTags in TestXML and TestHTML.